Firestore/Example/Tests/Integration/API/FIRArrayTransformTests.mm

```diff
@@ -66,20 +66,12 @@ - (void)tearDown {
 
 /** Waits for a snapshot with local writes. */
 - (FIRDocumentSnapshot *)waitForLocalEvent {
-  FIRDocumentSnapshot *snapshot;
-  do {
-    snapshot = [_accumulator awaitEventWithName:@"Local event."];
-  } while (!snapshot.metadata.hasPendingWrites);
-  return snapshot;
+  return [_accumulator awaitLocalEventWithName:@"Local event."];
```

Comment:
These function don't take arguments in the JS repo (additionally, the name that's passed here doesn't really make this more descriptive). Can we remove the name argument?

Reply:
I would be fine with that...  but then that begs the question why the existing "awaitEventWithName" requires a name but "awaitLocalEvent" and "awaitRemoteEvent" do not...  So it seems like we should perhaps also remove the name from awaitEvent to make FSTEventAccumulator consistent with itself...  but presumably whoever added it did it intentionally and removing it would also require changing a ton of tests.  So I'm not exactly sure where to land with this PR...  :-/  Any strong feelings?

Reply:
My answer to that would just be that `awaitLocalEvent` calls `awaitEvent:Local Event` and `awaitRemoteEvent` calls it with `Remote Event`. None of this is strictly needed, but it would reduce some of my confusion, and not require any major changes.

Reply:
Fair enough.  I reworked this as suggested, and also made FSTEventAccumulator generic, like it is on Web and Android (was necessary to avoid some casting on the return value of awaitLocalEvent).  Take a look if you like, else I'll probably submit once travis completes.

Reply:
Looks good. Thanks!